File: relisoft-hr/server/models/LeaveType.js
import mongoose from 'mongoose';

const leaveTypeSchema = new mongoose.Schema({
  name: { type: String, required: true, unique: true },
  code: { type: String, unique: true, uppercase: true },
  description: { type: String },
  isPaid: { type: Boolean, default: true },
  isCompOff: { type: Boolean, default: false },
  isFloater: { type: Boolean, default: false },
  requiresAdvanceNotice: { type: Boolean, default: false },
  advanceNoticeDays: { type: Number, default: 0 },
  isAccrued: { type: Boolean, default: false },
  accrualPerMonth: { type: Number, default: 0 },
  maxConsecutiveDays: { type: Number },
  requiresDocumentation: { type: Boolean, default: false },
  colorCode: { type: String, default: '#6366f1' },
  isActive: { type: Boolean, default: true },
}, { timestamps: true });

export default mongoose.model('LeaveType', leaveTypeSchema);

File: relisoft-hr/server/Models/LeaveType.cs
using System.ComponentModel.DataAnnotations;

namespace RelisoftHR.Models;

public class LeaveType
{
    [Key]
    public int Id { get; set; }
    [Required, MaxLength(100)]
    public string Name { get; set; } = "";
    public bool IsActive { get; set; } = true;
    public int SortOrder { get; set; }
    public decimal CarryForwardPct { get; set; }
    public bool IsCompOff { get; set; }
    public bool IsFloaterHoliday { get; set; }
    public int MaxFloaterPerYear { get; set; }
    public int CompOffValidityDays { get; set; }
}
